org/lateralgm/subframes/SpriteFrame.java

```diff
@@ -1269,6 +1268,10 @@ else if (cmd.endsWith(".PASTE"))
 					LGM.showDefaultExceptionHandler(e);
 					}
 				imageChanged = true;
+				for (int i = 0; i < images.bi.size(); i++)
+					{
+					images.bi.set(i, Util.cloneImage(images.bi.get(i)));
```

Comment:
You are discarding the original copy of these images. So even the first time you go to make a paste, you're copying them. Please just call `res.subImages.add(Util.cloneImage(images.bi.get(i)));`.

Reply:
Good point.